tensorflow_addons/rnn/cell.py

```diff
@@ -366,3 +366,425 @@ def _create_norm_layer(self, name):
+            W_hh : Weights for prev. outputs y_{t-1} (`self.recurrent_kernel`)
+            b : Bias term for centering (`self.bias`)
+            d1 : Dropout function for x_t (`inputs * dp_mask`)
+            d2 : Dropout function for y_{t-1} (`prev_output * rec_dp_mask`)
+            ln : Scaling function from layer normalization (`self.layernorm`)
+            f : Activation function (`self.activation`)
+
+        Case 1:
+            Simple RNN, only with bias and activation
+              y_t = f(x_t * W_xh + y_{t-1} * W_hh + b)
+            or
+              net = x_t * W_xh + y_{t-1} * W_hh
+              y_t = f(net + b)
+
+        Case 2:
+            RNN with, layer normalization (only scaling), bias and activation.
+              y_t = f(ln(x_t * W_xh + y_{t-1} * W_hh) + b)
+            or
+              net = x_t * W_xh + y_{t-1} * W_hh
+              y_t = f(ln(net) + b)
+
+            Layer normalization with scaling and centering in one go (see Ba et
+            al (2016), page 3, formula 4, https://arxiv.org/abs/1607.06450)
+            is the same as layer normalization only with scaling, and
+            centering directly afterwards.
+
+        Case 3:
+            RNN, with dropout, bias, and activation (no scaling from LN)
+              y_t = f(d1(x_t) * W_xh + d2(y_{t-1}) * W_hh + b)
+            or
+              net = d1(x_t) * W_xh + d2(y_{t-1}) * W_hh
+              y_t = f(net + b)
+
+        Case 4:
+            Everyting is used, i.e. all dropouts, layer normalization
+            (only scaling), bias, and activation
+              y_t = f(ln(d1(x_t) * W_xh + d2(y_{t-1}) * W_hh) + b)
+            or
+              net = d1(x_t) * W_xh + d2(y_{t-1}) * W_hh
+              y_t = f(ln(net) + b)
+        """
+        prev_output = states[0]
+        dp_mask = self.get_dropout_mask_for_cell(inputs, training)
+        rec_dp_mask = self.get_recurrent_dropout_mask_for_cell(
+            prev_output, training)
+
+        if dp_mask is not None:
+            h = keras.backend.dot(inputs * dp_mask, self.kernel)
+        else:
+            h = keras.backend.dot(inputs, self.kernel)
+
+        # don't add bias to "h" here
+        # add bias after scaling with layer normalization to "output"
+
+        if rec_dp_mask is not None:
+            prev_output = prev_output * rec_dp_mask
+        output = h + keras.backend.dot(prev_output,
+                                       self.recurrent_kernel)  # "net"
+
+        output = self.layernorm(output)
+
+        if self.bias is not None:
+            output = keras.backend.bias_add(output, self.bias)
+
+        if self.activation is not None:
+            output = self.activation(output)
+
+        return output, [output]
+
+    # use SimpleRNNCell's get_initial_state method
+
+    def get_config(self):
+        cell_config = super(LayerNormSimpleRNNCell, self).get_config()
+        del cell_config['name']
+
+        ln_config = self.layernorm.get_config()
+        ln_config = {
```

Comment:
How about this?

```
ln_config = {k:v for k, v in ln_config.items()
                    if k in ["epsilon", "gamma_initializer", "gamma_regularizer", "gamma_constraint"]}
```

Reply:
yes. your's looks more pythonic
https://github.com/tensorflow/addons/pull/841/commits/dc22b6e85e1a6bf5a8412a26b0b196bd35e504c5